lib/common.go

```diff
@@ -187,16 +199,20 @@ func BigIntToBytes(alph Alphabet, x *big.Int, l int) []byte {
 
 	// the length must be the same as the original byte
 	// so we add extra zeros to the end if necessary
+	overflow := false
 	if len(res) < l {
 		res = append(res, make([]byte, l-len(res))...)
+	} else if len(res) > l {
+		overflow = true
+		res = res[0:l]
 	}
 
 	// convert the byte slice to string readable
 	for i := range res {
 		res[i] = Alphabets[alph][res[i]]
```

Comment:
This for-loop can also be moved to a separate function for better SRP compliance.